src/input/isf.c

```diff
@@ -287,12 +286,16 @@ static int parse_isf_header(GString *buf, struct context *inc)
 
 	if (inc == NULL)
 		return SR_ERR_ARG;
-	
+
+	/* Search for all header items. */
 	for (i = 0; i < HEADER_ITEMS_PARAMETERS; ++i) {
 		pattern = find_item(buf, header_items[i]);
-		if (pattern == NULL)
-			/* Ignore unknown command/header.  */
-			continue;
+		if (pattern == NULL) {
+			/* WFID is not required. */
+			if (i == WFID)
+				continue;
```

Comment:
For me (.isf created by Tektronix DPO 4034) "WFMTYPE" was the problem, not "WFID", as could be determined from the .isf header I had posted:
https://github.com/sigrokproject/libsigrok/pull/186#discussion_r1083373879

The change in your previous commit (the removed lines above had been added there), which ignored all unknown header items (proposal from my review), worked for me. After this change it didn't work anymore.

Reply:
The thing is that it is pretty hard to identify which header items should be mandatory as the header slightly varies depending on the device used.

Do you think it would be better to initialize header items to default values and make them all optional?

Reply:
I'm not sure because I didn't had a deep look at some specifications. But I don't think there are sensible default values possible in most cases.

I couldn't find WFMTYPE in the manual linked in the comment at the beginning of isf.c file.

Reply:
The default option for WFMTYPE could be `ANALOG` value. The other options are probably rarely used since they are used for radio frequency data. The options are specified in this [manual](https://download.tek.com/manual/MSO3000-and-DPO3000-Programmer-Manual.pdf) on page 2-586.